### PR TITLE
Generally cleans up the FEA_ files and adds dmdoc.

### PR DESCRIPTION
**Pull request by Laboredih123**
[ATMOS][INTERNAL][CLEAN]
## About the PR 
Changes the FEA_ files and machinery.dm to absolute pathing, optimises some things, adds dmdoc, and betters readability.

## Why's this needed? 
Atmos code become cleaner is good. It staying ugly is bad.

### CODE REVIEW

File: code/modules/atmospherics/FEA_turf_tile.dm [hunk 1]
```diff
@@ -76,6 +77,7 @@ var/global/list/turf/hotly_processed_turfs = list()
 		pressure_difference = connection_difference
 		pressure_direction = connection_direction
 
+///
```
Comment (Tarmunora) [inline code/modules/atmospherics/FEA_turf_tile.dm:80]: ?



</p>
</details>

File: code/modules/atmospherics/FEA_turf_tile.dm [hunk 2]
```diff
@@ -592,50 +591,50 @@ var/global/list/turf/hotly_processed_turfs = list()
...
 				air_master.tiles_to_update |= south
-		if(istype(east))
+		if(east.turf_flags & IS_TYPE_SIMULATED)
 			east.tilenotify(src)
 			if(east.parent)
 				air_master.groups_to_rebuild |= east.parent
 			else
 				air_master.tiles_to_update |= east
-		if(istype(west))
+		if(west.turf_flags & IS_TYPE_SIMULATED)
```
Comment (Tarmunora) [inline code/modules/atmospherics/FEA_turf_tile.dm:619]: Hm. What does get_step return at the edge of the map? If it returns null, need to use `?.` on those var accesses or this would runtime in that edge case

File: code/modules/atmospherics/FEA_turf_tile.dm [hunk 3]
```diff
@@ -592,50 +591,50 @@ var/global/list/turf/hotly_processed_turfs = list()
...
-		if(istype(south))
+		if(south.turf_flags & IS_TYPE_SIMULATED)
 			south.tilenotify(src)
 			air_master.tiles_to_update |= south
-		if(istype(east))
+		if(east.turf_flags & IS_TYPE_SIMULATED)
 			east.tilenotify(src)
 			air_master.tiles_to_update |= east
-		if(istype(west))
+		if(west.turf_flags & IS_TYPE_SIMULATED)
```
Comment (Tarmunora) [inline code/modules/atmospherics/FEA_turf_tile.dm:637]: ditto

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/modules/atmospherics/machinery.dm [hunk 4]
```diff
@@ -1,62 +1,52 @@
...
+ABSTRACT_TYPE(/obj/machinery/atmospherics)
+
+/**
+ * Quick overview:
+ * Pipes combine to form pipelines.
+ * Pipelines and other atmospheric objects combine to form pipe_networks.
+ * Note: A single pipe_network represents a completely open space.
+ *
+ * Pipes -> Pipelines
+ * Pipelines + Other Objects -> Pipe network
```
Comment (TheTyrant25) [inline code/modules/atmospherics/machinery.dm:10]: is this block doc comment attached to `ABSTRACT_TYPE(/obj/machinery/atmospherics)`? It looks like it should be but it might be worth double checking.

File: code/modules/atmospherics/machinery.dm [hunk 5]
```diff
@@ -1,62 +1,52 @@
...
-
-			return null
+/obj/machinery/atmospherics/process()
+	build_network()
+	..()
 
-		network_expand(datum/pipe_network/new_network, obj/machinery/atmospherics/pipe/reference)
-			// Check to see if should be added to network. Add self if so and adjust variables appropriately.
-			// Note don't forget to have neighbors look as well!
+// override default subscribes to be in a different process loop. that's why they don't call parent ( ..() )
```
Comment (TheTyrant25) [inline code/modules/atmospherics/machinery.dm:21]: doc comment

File: code/modules/atmospherics/machinery.dm [hunk 6]
```diff
@@ -1,62 +1,52 @@
...
+/// Check to see if should be added to network. Add self if so and adjust variables appropriately.
+/// Note don't forget to have neighbors look as well!
+/obj/machinery/atmospherics/proc/network_expand(datum/pipe_network/new_network, obj/machinery/atmospherics/pipe/reference)
 
-		return_network(obj/machinery/atmospherics/reference)
-			// Returns pipe_network associated with connection to reference
-			// Notes: should create network if necessary
-			// Should never return null
+/// Called to build a network from this node.
+/obj/machinery/atmospherics/proc/build_network()
```
Comment (TheTyrant25) [inline code/modules/atmospherics/machinery.dm:37]: wait so is there just no code in this proc?

File: code/modules/atmospherics/machinery.dm [hunk 7]
```diff
@@ -1,62 +1,52 @@
...
-			// Used when two pipe_networks are combining
+/// Used when two pipe_networks are combining.
+/obj/machinery/atmospherics/proc/reassign_network(datum/pipe_network/old_network, datum/pipe_network/new_network)
 
-		return_network_air(datum/pipe_network/reference)
-			// Return a list of gas_mixture(s) in the object
-			//		associated with reference pipe_network for use in rebuilding the networks gases list
-			// Is permitted to return null
+/// Return a list of gas_mixture(s) in the object
+///	associated with reference pipe_network for use in rebuilding the networks gases list.
```
Comment (TheTyrant25) [inline code/modules/atmospherics/machinery.dm:47]: block doc comment pls. Also weird tabbing on that second line.

File: code/modules/atmospherics/FEA_turf_tile.dm (hunk 1 above)
Comment (TheTyrant25) [reply code/modules/atmospherics/FEA_turf_tile.dm:80]: the doc comment'nt

File: code/modules/atmospherics/FEA_gas_mixture.dm [hunk 8]
```diff
@@ -4,19 +4,27 @@ What are the archived variables for?
...
+
 /datum/gas/oxygen_agent_b
 	specific_heat = 300
 
-
+/**
+ * The key of atmospherics.
+ * This datum here is how we represent gas mixtures.
+ * Temperature is in Kelvin and volume is in Litres.
+ * Todo cringe: write more about gas mixtures
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_gas_mixture.dm:26]: to people who don't know you're called cringe, this can be confusing. Try "cringe's todo:" or something. Because cringe is an adjective by itself and stuff. Yeah.

File: code/modules/atmospherics/FEA_gas_mixture.dm [hunk 9]
```diff
@@ -64,6 +80,7 @@ What are the archived variables for?
 		nitrogen = MOLES_N2STANDARD * 0.5
 		temperature = OCEAN_TEMP
 
+/// Completely removes any gas.
 /datum/gas_mixture/proc/vacuum() //yknow, for when you want "zero" to actually mean "zero".
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_gas_mixture.dm:84]: maybe move this comment to the doc comment or remove it.

File: code/modules/atmospherics/FEA_gas_mixture.dm [hunk 10]
```diff
@@ -203,35 +222,34 @@ What are the archived variables for?
...
 	ARCHIVED(temperature) = temperature
 	graphic_archived = graphic
-	return 1
 #endif
 
-//Similar to merge(...) but first checks to see if the amount of air assumed is small enough
-//	that group processing is still accurate for source (aborts if not)
-//Returns: 1 on successful merge, 0 if the check failed
+/// Similar to [/datum/gas_mixture/proc/merge] but first checks to see if the amount of air assumed is small enough
+///	that group processing is still accurate for source (aborts if not).
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_gas_mixture.dm:228]: doc block comment pls

File: code/modules/atmospherics/FEA_gas_mixture.dm [hunk 11]
```diff
@@ -310,22 +328,22 @@ What are the archived variables for?
 
 	return removed
 
-//Similar to remove(...) but first checks to see if the amount of air removed is small enough
-//	that group processing is still accurate for source (aborts if not)
-//Returns: gas_mixture with the gases removed or null
+/// Similar to [/datum/gas_mixture/proc/remove] but first checks to see if the amount of air removed is small enough
+/// that group processing is still accurate for source (aborts if not).
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_gas_mixture.dm:332]: doc block comment

File: code/modules/atmospherics/FEA_fire.dm [hunk 12]
```diff
@@ -91,196 +90,200 @@
...
 #endif
 
-	animate_movement = NO_STEPS // fix for weird unpool sliding
-
 	var/volume = 125
 	var/temperature = FIRE_MINIMUM_TEMPERATURE_TO_EXIST
-	var/just_spawned = 1
-	var/bypassing = 0
+	/// If we've just spawned, don't die right after, wait a cycle.
+	var/just_spawned = FALSE
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_fire.dm:96]: You seem to have changed just_spawned from 1 to `FALSE`. Will that break things (is it supposed to be `TRUE`) or is that intended?

File: code/modules/atmospherics/FEA_fire.dm [hunk 13]
```diff
@@ -91,196 +90,200 @@
...
+	var/input = temperature / 100
+
+	var/red
+	if (input <= 66)
+		red = 255
+	else
+		red = input - 60
+		red = 329.698727446 * (red ** -0.1332047592)
+	red = clamp(red, 0, 255)
+
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_fire.dm:133]: The diff here in this proc is quite large. I'm glossing over this bit and hoping that apart from the indentation and doc comment, nothing much changed.

File: code/modules/atmospherics/FEA_fire.dm [hunk 14]
```diff
@@ -91,196 +90,200 @@
...
-		blue = clamp(blue, 0, 255)
+			blue = input - 10
+			blue = 138.5177312231 * log(blue) - 305.0447927307
+	blue = clamp(blue, 0, 255)
 
-		color = rgb(red, green, blue) //changing obj.color is not expensive, and smooths imperfect light transitions
+	color = rgb(red, green, blue) //changing obj.color is not expensive, and smooths imperfect light transitions
 
-		// dear lord i apologuize for this conditional which i am about to write and commit. please be with the starving pygmies down in new guinea amen
+	// dear lord i apologuize for this conditional which i am about to write and commit. please be with the starving pygmies down in new guinea amen
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_fire.dm:156]: apologise*

File: code/modules/atmospherics/FEA_fire.dm [hunk 15]
```diff
@@ -91,196 +90,200 @@
...
+	var/list/curc = medium_light_rgbas?["hotspot"] || list(0, 0, 0, 100)
+	var/red_mean = ((red + curc[1]) /2) // mean of R components in the two compared colors
+	var/deltaR2 = (red   - (curc[1]))**2
+	var/deltaG2 = (blue  - (curc[2]))**2
+	var/deltaB2 = (green - (curc[3]))**2
 #endif
 
-		//this is our weighted euclidean distance function, weights based on red component
-		var/color_delta = ( (((512+red_mean)*(deltaR2**2))>>8) + (4*(deltaG2**2)) + (((767-red_mean)*(deltaB2**2))>>8) )
+	//this is our weighted euclidean distance function, weights based on red component
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_fire.dm:175]: possibly doc comment this

File: code/modules/atmospherics/FEA_system.dm [hunk 16]
```diff
@@ -1,322 +1,280 @@
...
+ * 	This prevents abnormalities like air/fire spreading rapidly in one direction and super slowly in the other.
+ *
+ * Why not just archive everything and then calculate?
+ * 	Efficiency. While a for-loop that goes through all tils and groups to archive their information before doing any calculations seems simple, it is
+ * 		slightly less efficient than the archive-before-modify/read method.
+ *
+ * Why is there a cycle check for calculating data as well?
+ * 	This ensures that every connection between group-tile, tile-tile, and group-group is only evaluated once per loop.
+ *
+ * 	atom/Cross(atom/movable/mover, turf/target, height, air_group)
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_system.dm:17]: im not reading all this lol. I trust you've kept it legible.

File: code/modules/atmospherics/FEA_system.dm [hunk 17]
```diff
@@ -1,322 +1,280 @@
...
+	var/list/turf/simulated/active_super_conductivity = list() //gets space in here somehow ZEWAKA/ATMOS
 	var/list/turf/simulated/high_pressure_delta = list()
 
-	//Geometry updates lists
+	/// Turfs that are in this list have their border data updated before the next air calculations for a cycle.
+	///Place turfs in this list rather than call the proc directly to prevent race conditions
 	var/list/turf/tiles_to_update = list()
+
+	/// A list of air groups that have had their geometry occluded and thus may need to be split in half.
+	///	A set of adjacent groups put in here will join together if validly connected.
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_system.dm:67]: doc block comment please

File: code/modules/atmospherics/FEA_system.dm [hunk 18]
```diff
@@ -1,322 +1,280 @@
...
-		is_busy = TRUE
-
-		if(!explosions.exploding)
-			if(groups_to_rebuild.len > 0)
-				process_rebuild_select_groups()
-			LAGCHECK(LAG_REALTIME)
+	var/turf/space/space_sample //instead of repeatedly using locate() to find space, we should just cache a space tile ok
+
+/// Updates cached space sample if need be.
+// Returns: New space sample.
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_system.dm:82]: doc comment oopsie

File: code/modules/atmospherics/FEA_system.dm [hunk 19]
```diff
@@ -1,322 +1,280 @@
...
+/datum/controller/air_system/proc/update_space_sample()
+	if (!space_sample || !(space_sample.turf_flags & CAN_BE_SPACE_SAMPLE))
+		space_sample = locate(/turf/space)
+	return space_sample
+
+/datum/controller/air_system/proc/setup(datum/controller/process/air_system/controller)
+	parent_controller = controller
+
+	#if SKIP_FEA_SETUP == 1
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_system.dm:91]: is `SKIP_FEA_SETUP` a bool btw? If it isn't, you might want to use defines or something. That or `#ifdef`.

File: code/modules/atmospherics/FEA_system.dm [hunk 20]
```diff
@@ -1,322 +1,280 @@
...
+		S.update_air_properties()
+
+	boutput(world, "<span class='alert'>Geometry processed in [(world.timeofday-start_time)/10] seconds!</span>")
+	#endif
+
+/// Collects turfs into groups.
+/datum/controller/air_system/proc/assemble_group_turf(turf/simulated/base)
+	set waitfor = 0
+	var/list/turf/simulated/members = list(base) // Confirmed group members
+	var/list/turf/simulated/possible_members = list(base) // Possible places for group expansion
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_system.dm:111]: maybe make these comments doc comments

File: code/modules/atmospherics/FEA_system.dm [hunk 21]
```diff
@@ -1,322 +1,280 @@
...
+			for(var/turf/simulated/possible as anything in possible_space_borders)
+				if (possible == test)
+					test.dist_to_space = 1
+					break
+				dist = GET_DIST(possible, test)
+				if (!test.dist_to_space || (dist < test.dist_to_space))
+					test.dist_to_space = dist
+
+		// Allow groups to determine if group processing is applicable after FEA setup
+		if(current_cycle) group.group_processing = FALSE
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_system.dm:162]: i know this works fine but the lack of indentation on lines 163 and 164 confused me for a second. Maybe put `group.group_processing = FALSE` into another line for readability.

**Review by TheTyrant25 — COMMENTED** [review]
Overally seems like a very nice and much needed cleanup. Some of the larger changes with like 20 lines in a row being swapped out confused the diff renderer a bit so they were harder to review.

File: code/modules/atmospherics/FEA_system.dm (hunk 20 above)
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_system.dm:111]: no can do, these are temp vars since they in a proc.

File: code/modules/atmospherics/machinery.dm (hunk 4 above)
Comment (Laboredih123) [reply code/modules/atmospherics/machinery.dm:10]: it is not attached, dont want it to be.

File: code/modules/atmospherics/machinery.dm (hunk 6 above)
Comment (Laboredih123) [reply code/modules/atmospherics/machinery.dm:37]: its a parent proc meant to be overriden

File: code/modules/atmospherics/machinery.dm (hunk 4 above)
Comment (TheTyrant25) [reply code/modules/atmospherics/machinery.dm:10]: in that case line 3 can remove one asterisk

File: code/modules/atmospherics/FEA_system.dm (hunk 20 above)
Comment (TheTyrant25) [reply code/modules/atmospherics/FEA_system.dm:111]: nevermind then

File: code/modules/atmospherics/FEA_gas_mixture.dm (hunk 8 above)
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_gas_mixture.dm:26]: its just temporary, will add more later

File: code/modules/atmospherics/FEA_fire.dm (hunk 14 above)
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_fire.dm:156]: zewaka was feeling really apologuietic you see

File: code/modules/atmospherics/FEA_fire.dm (hunk 12 above)
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_fire.dm:96]: ah its not intended, fixing

File: code/modules/atmospherics/machinery.dm (hunk 6 above)
Comment (TheTyrant25) [reply code/modules/atmospherics/machinery.dm:37]: i see

File: code/modules/atmospherics/FEA_fire.dm (hunk 14 above)
Comment (TheTyrant25) [reply code/modules/atmospherics/FEA_fire.dm:156]: understandable. yet

File: code/modules/atmospherics/FEA_fire.dm (hunk 15 above)
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_fire.dm:175]: cant, inside proc

File: code/modules/atmospherics/FEA_system.dm (hunk 19 above)
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_system.dm:91]: it reaches into other files so not touching

Comment (TheTyrant25) [reply code/modules/atmospherics/FEA_system.dm:91]: fair

File: code/modules/atmospherics/machinery.dm (hunk 5 above)
Comment (Laboredih123) [reply code/modules/atmospherics/machinery.dm:21]: no need, doesnt do much special except subscribe to a different loop instead of the main loop. this just explains to coders why parent isnt called

Comment (TheTyrant25) [reply code/modules/atmospherics/machinery.dm:21]: ah alrighty then

File: code/modules/atmospherics/FEA_airgroup.dm [hunk 22]
```diff
@@ -225,10 +226,7 @@
 						if(connection_difference > 0)
 							self_border.consider_pressure_difference(connection_difference, get_dir(self_border,enemy_border))
 						else
-							var/turf/enemy_turf = enemy_border
-							if(!isturf(enemy_turf))
-								enemy_turf = enemy_border.loc
-							enemy_turf.consider_pressure_difference(-connection_difference, get_dir(enemy_turf,self_border))
+							enemy_border.consider_pressure_difference(-connection_difference, get_dir(enemy_border,self_border))
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_airgroup.dm:229]: the amount of indents on this line is causing it to show up weird in github. Looks correct tho.

File: code/modules/atmospherics/FEA_airgroup.dm [hunk 23]
```diff
@@ -269,11 +266,6 @@
 				if(connection_difference)
 					if(connection_difference > 0 && !isnull(self_border))
 						self_border.consider_pressure_difference(connection_difference, get_dir(self_border,enemy_tile))
-					else
-						var/turf/enemy_turf = enemy_tile
-						if(!isturf(enemy_turf))
-							enemy_turf = enemy_tile.loc
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_airgroup.dm:273]: why was this removed btw?

File: code/modules/atmospherics/FEA_fire.dm [hunk 24]
```diff
@@ -93,8 +93,10 @@
 	var/volume = 125
 	var/temperature = FIRE_MINIMUM_TEMPERATURE_TO_EXIST
 	/// If we've just spawned, don't die right after, wait a cycle.
-	var/just_spawned = FALSE
+	var/just_spawned = TRUE
+	///
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_fire.dm:97]: ?

File: code/modules/atmospherics/FEA_gas_mixture.dm [hunk 25]
```diff
@@ -358,9 +359,7 @@ What are the archived variables for?
 
 	temperature = sample.temperature
 
-	return TRUE
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_gas_mixture.dm]: was this important?

File: code/modules/atmospherics/FEA_gas_mixture.dm [hunk 26]
```diff
@@ -371,11 +370,10 @@ What are the archived variables for?
 			var/datum/gas/corresponding = src.get_or_add_trace_gas_by_type(trace_gas.type)
 			corresponding.moles -= trace_gas.moles
 
-	return TRUE
-
 /// Checks if us and the sharer have a low enough delta and our combination's delta is small enough that group processing can be preserved.
-/// Returns: 0 if the self-check failed then -1 if sharer-check failed then 1 if both checks pass.
+/// Returns: 0 if the self-check failed, -1 if sharer-check failed, 1 if both checks pass.
 /datum/gas_mixture/proc/check_gas_mixture(datum/gas_mixture/sharer)
+	. = 1
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_gas_mixture.dm:376]: i confess to being unsure what this does

Comment (TheTyrant25) [reply code/modules/atmospherics/FEA_gas_mixture.dm:376]: also should it be TRUE instead of 1?

File: code/modules/atmospherics/FEA_turf_tile.dm [hunk 27]
```diff
@@ -150,6 +182,7 @@ var/global/list/turf/hotly_processed_turfs = list()
 			qdel(gas_icon_overlay)
 			gas_icon_overlay = null
 
+
```
Comment (TheTyrant25) [inline code/modules/atmospherics/FEA_turf_tile.dm:185]: ?

File: code/modules/atmospherics/machinery.dm [hunk 28]
```diff
@@ -7,15 +7,14 @@ ABSTRACT_TYPE(/obj/machinery/atmospherics)
  * Note: A single pipe_network represents a completely open space.
  *
```
Comment (TheTyrant25) [inline code/modules/atmospherics/machinery.dm:8]: Could probably ditch this empty block comment line if you like.

**Review by TheTyrant25 — APPROVED** [review]
Overall much better clarity and commenting than in the first draft. Looks good to go to me, aside from extremely minor nitpicks.

**Comment by Laboredih123** [comment]
epic

**Comment by Laboredih123** [comment]
wrong button fuck

File: code/modules/atmospherics/FEA_airgroup.dm (hunk 23 above)
Comment (TheTyrant25) [reply code/modules/atmospherics/FEA_airgroup.dm:273]: fair

File: code/modules/atmospherics/FEA_gas_mixture.dm (hunk 26 above)
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_gas_mixture.dm:376]: i didnt like the return true at the end, albeit it doesnt make much sense now. also i cant use true since this function has 3 states, not 2.

File: code/modules/atmospherics/FEA_gas_mixture.dm (hunk 25 above)
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_gas_mixture.dm]: never used but eh, could i guess

File: code/modules/atmospherics/FEA_gas_mixture.dm (hunk 26 above)
Comment (TheTyrant25) [reply code/modules/atmospherics/FEA_gas_mixture.dm:376]: if its got three states maybe use defines instead of magic numbers

Comment (Laboredih123) [reply code/modules/atmospherics/FEA_gas_mixture.dm:376]: idk where to put the defines.
atmos.dm?

File: code/modules/atmospherics/FEA_turf_tile.dm (hunk 27 above)
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_turf_tile.dm:185]: accidental newline, removing

File: code/modules/atmospherics/machinery.dm (hunk 28 above)
Comment (Laboredih123) [reply code/modules/atmospherics/machinery.dm:8]: okee

File: code/modules/atmospherics/FEA_gas_mixture.dm (hunk 26 above)
Comment (TheTyrant25) [reply code/modules/atmospherics/FEA_gas_mixture.dm:376]: in atmos.dm or somewhere in _std/defines folder i suppose

File: code/modules/atmospherics/FEA_fire.dm [hunk 29]
```diff
@@ -1,79 +1,83 @@
...
 
-
-
-/turf/simulated/hotspot_expose(exposed_temperature, exposed_volume, soh, electric = 0)
-	. = ..()
-	var/datum/gas_mixture/air_contents = return_air()
-
+/// * Checks if we should light on fire if we do not have a hotspot already. If we should and don't have one, spawns one.
+/// * Returns: TRUE if we ignited or already have a hotspot, FALSE if we didn't make one or have one.
+/turf/simulated/hotspot_expose(exposed_temperature, exposed_volume, soh, electric = FALSE)
```
Comment (TobleroneSwordfish) [inline code/modules/atmospherics/FEA_fire.dm:29]: I've asked dev chat if anyone knows what `soh` means by the way, I have no clue lol

File: code/modules/atmospherics/FEA_gas_mixture.dm [hunk 30]
```diff
@@ -151,10 +169,12 @@ What are the archived variables for?
 		. |= REACTION_ACTIVE
 
 	fuel_burnt = 0
-	if(temperature > FIRE_MINIMUM_TEMPERATURE_TO_EXIST)
-		if(fire() > 0)
+	if(src.temperature > FIRE_MINIMUM_TEMPERATURE_TO_EXIST)
+		if(fire())
```
Comment (TobleroneSwordfish) [inline code/modules/atmospherics/FEA_gas_mixture.dm:173]: don't trust that this won't randomly become negative and ruin your day
```suggestion
		if(fire() > 0)
```

Comment (Laboredih123) [reply code/modules/atmospherics/FEA_gas_mixture.dm:173]: better idea to runtime if it goes below 0 because that means we have burnt negative moles, which uh, bad

File: code/modules/atmospherics/FEA_fire.dm (hunk 29 above)
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_fire.dm:29]: i have a faint idea on what it does and even fainter idea on why it does it. i have no idea what to remain soh to.

File: code/modules/atmospherics/FEA_airgroup.dm [hunk 31]
```diff
@@ -153,9 +158,7 @@
 					// Tiles can get added to these lists more than once, but that is OK,
 					// because groups sharing more than one edge should transfer more air.
 
-					//if(istype(enemy_tile) && enemy_tile.parent && enemy_tile.parent.group_processing) //trying the other one
-					if(enemy_tile.turf_flags & IS_TYPE_SIMULATED && enemy_tile.parent && enemy_tile.parent.group_processing) //blahh danger
-
+					if(issimulatedturf(enemy_tile) && enemy_tile.parent && enemy_tile.parent.group_processing) //blahh danger
```
[the remark this answers is not in the document]
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_airgroup.dm:161]: it goes into other files that are not fea and i dont want to risk my atomisation.

File: code/modules/atmospherics/FEA_fire.dm (hunk 29 above)
Comment (TheTyrant25) [reply code/modules/atmospherics/FEA_fire.dm:29]: sounds like "standard of heating" or suchlike. h could be hotspot too. really poor naming lol

File: code/modules/atmospherics/FEA_gas_mixture.dm (hunk 30 above)
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_gas_mixture.dm:173]: ill put an assert

File: code/modules/atmospherics/FEA_airgroup.dm (hunk 23 above)
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_airgroup.dm:273]: it also did absolutely nothing because it inputted a dir of 0

Comment (Laboredih123) [reply code/modules/atmospherics/FEA_airgroup.dm:273]: it was a tile pointing to itself

Comment (Laboredih123) [reply code/modules/atmospherics/FEA_airgroup.dm:273]: i removed it later on

File: code/modules/atmospherics/FEA_fire.dm (hunk 29 above)
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_fire.dm:29]: someone said set own hotspot but im not sure how accurate that is

Comment (Laboredih123) [reply code/modules/atmospherics/FEA_fire.dm]: think we got it down in imcoder, renaming and resolving

File: code/modules/atmospherics/FEA_airgroup.dm (hunk 23 above)
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_airgroup.dm:273]: Think i can avoid checking if its nonsimulated by making sure that cant ever happen in normal gameplay. replacewith only rebuilds groups if the new tile is simulated but doesnt rebuild if the old tile was simulated and changed into a nonsimulated one. ill do that in a new pr.

Comment (Laboredih123) [reply code/modules/atmospherics/FEA_airgroup.dm:273]: leaving an assert istype in every for loop could get expensive

Comment (TobleroneSwordfish) [reply code/modules/atmospherics/FEA_airgroup.dm:273]: It already had an istype check, either put the code back or put an assert, don't add a silent failure.

File: code/modules/atmospherics/FEA_gas_mixture.dm (hunk 30 above)
Comment (TobleroneSwordfish) [reply code/modules/atmospherics/FEA_gas_mixture.dm:173]: You didn't put an assert

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

File: code/modules/atmospherics/FEA_airgroup.dm (hunk 23 above)
Comment (Laboredih123) [reply code/modules/atmospherics/FEA_airgroup.dm:273]: putting a stacktrace instead, shouldn't be problematic enough to warrant a crash but will rebuild group. previous code did nothing useful at all since it had a dir of 0 inputted.
